src/sentry/utils/snuba.py

```diff
@@ -298,13 +298,67 @@ def transform_aliases_and_query(**kwargs):
     conditions = kwargs['conditions'] or []
     filter_keys = kwargs['filter_keys']
 
+    def parse_selected_column_condition(column):
+        new_column = [column[0]]
+        new_condition = []
+
+        if column[0] == 'if':
+            if_condition = column[1][0]
+            new_if_condition = []
+
+            if isinstance(if_condition, list) and if_condition[0] == 'in':
```

Comment:
This seems really special-purpose and fragile. I think it should be possible to use an approach similar to https://github.com/getsentry/snuba/blob/d962a354338abc29f9331c927d16ba5e560f1d59/snuba/util.py#L281-L296 to find column names inside nested functions and translate them in-place there.

Reply:
Yep agreed - I couldn't think of a way to make this generic enough. One issue is that we assume if a column name is not in `SENTRY_SNUBA_MAP` then we transform it: https://github.com/getsentry/sentry/blob/3fd534be6b269f5b6fe58ac8b9e669c64f8ad729/src/sentry/utils/snuba.py#L270

Also our "col in list" expression has a different format afaict, where it's ["in", [<column name>, "tuple", <list>]] where it seems like most expressions treat the first item in a list as a keyword and the rest as arguments?